include/util/coordinate.hpp

```diff
@@ -40,7 +40,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 namespace osrm
 {
 
-constexpr const double COORDINATE_PRECISION = 1e6;
+constexpr const double COORDINATE_PRECISION = 1e7;
```

Comment:
@daniel-j-h it is better to cleanup public/private headers first. Anyway i think better to split PR into two commits:
* change to rounding coordinates to reduce error in coordinate projection with transformations float-point -> fixed-point -> float-point -> fixed-point. The added regression test expects `to(from(x)) = x`, but only for some specific locations.
* change to 1e7 precision
The first change is in the PR, but the latter change is more complicated to handle because it hardly influences computations in guidance, i hope in better direction, but all failing tests must be analyzed as a separate PR